airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```diff
@@ -201,15 +202,17 @@ private void resetCheckpointValues() {
    *
    * @return {@link AirbyteStateMessage} which includes offset and schema history if used.
    */
-  private AirbyteMessage createStateMessage(final Map<String, String> offset) {
+  private AirbyteMessage createStateMessage(final Map<String, String> offset, final long recordCount) {
     if (trackSchemaHistory && schemaHistoryManager == null) {
       throw new RuntimeException("Schema History Tracking is true but manager is not initialised");
     }
     if (offsetManager == null) {
       throw new RuntimeException("Offset can not be null");
     }
 
-    return cdcStateHandler.saveState(offset, schemaHistoryManager != null ? schemaHistoryManager.read() : null);
+    final AirbyteMessage message = cdcStateHandler.saveState(offset, schemaHistoryManager != null ? schemaHistoryManager.read() : null);
+    message.getState().withSourceStats(new AirbyteStateStats().withRecordCount((double) recordCount));
```

Comment:
curious why is recordCount a double? seems like it can only be an integer

Reply:
protocol defined this as a  "number" in Json, which translated to double value in java thus here comes casting.